src/main/java/eu/europeana/metis/sandbox/config/IndexingConfig.java

```diff
@@ -20,62 +20,124 @@
+
+  @Value("${sandbox.publish.mongo.ssl-enable:#{null}}")
+  private Boolean mongoPublishEnableSSL;
+
+  @Value("${sandbox.publish.mongo.db}")
+  private String mongoPublishDb;
+
+  @Value("${sandbox.publish.solr.hosts}")
+  private String[] solrPublishHosts;
+
+  @Value("${sandbox.publish.solr.zookeeper.hosts:#{null}}")
+  private String[] zookeeperPublishHosts;
+
+  @Value("${sandbox.publish.solr.zookeeper.ports:#{null}}")
+  private int[] zookeeperPublishPorts;
+
+  @Value("${sandbox.publish.solr.zookeeper.chroot:#{null}}")
+  private String zookeeperPublishChroot;
+
+  @Value("${sandbox.publish.solr.zookeeper.default.collection:#{null}}")
+  private String zookeeperPublishDefaultCollection;
+
+  @Value("${sandbox.publish.solr.zookeeper.timeout:#{null}}")
+  private Integer zookeeperPublishTimeoutInSecs;
+
+  @Bean
+  Indexer previewIndexer()
+      throws URISyntaxException, SetupRelatedIndexingException, IndexerRelatedIndexingException {
+    return getIndexer(mongoPreviewHosts, mongoPreviewPorts, mongoPreviewDb,
+        mongoPreviewAuthenticationDb,
+        mongoPreviewUsername, mongoPreviewPassword, mongoPreviewEnableSSL, solrPreviewHosts,
+        zookeeperPreviewHosts, zookeeperPreviewPorts, zookeeperPreviewChroot,
+        zookeeperPreviewDefaultCollection, zookeeperPreviewTimeoutInSecs);
+  }
 
   @Bean
-  Indexer indexer()
+  Indexer publishIndexer()
       throws URISyntaxException, SetupRelatedIndexingException, IndexerRelatedIndexingException {
+    return getIndexer(mongoPublishHosts, mongoPublishPorts, mongoPublishDb,
+        mongoPublishAuthenticationDb,
+        mongoPublishUsername, mongoPublishPassword, mongoPublishEnableSSL, solrPublishHosts,
+        zookeeperPublishHosts, zookeeperPublishPorts, zookeeperPublishChroot,
+        zookeeperPublishDefaultCollection, zookeeperPublishTimeoutInSecs);
+  }
+
+  //Suppress: Methods should not have too many parameters warning
+  //We are okay with this method to ease configuration
+  @SuppressWarnings("squid:S107")
+  private Indexer getIndexer(String[] mongoHosts, int[] mongoPorts, String mongoDb,
+      String mongoAuthenticationDb, String mongoUsername, String mongoPassword,
+      Boolean mongoEnableSSL, String[] solrHosts, String[] zookeeperHosts, int[] zookeeperPorts,
+      String zookeeperChroot, String zookeeperDefaultCollection, Integer zookeeperTimeoutInSecs)
+      throws SetupRelatedIndexingException, URISyntaxException, IndexerRelatedIndexingException {
     checkArgument(isNotBlank(mongoDb), "Mongo db must be provided");
     checkArgument(isNotEmpty(mongoHosts), "Mongo hosts must be provided ");
     checkArgument(isNotEmpty(mongoPorts), "Mongo ports must be provided ");
     checkArgument(isNotEmpty(solrHosts), "Solr hosts must be provided ");
 
     var settings = new IndexingSettings();
 
-    if (isNull(mongoEnableSSL)) {
-      mongoEnableSSL = Boolean.FALSE;
-    }
+    boolean enableSSL = !isNull(mongoEnableSSL) && mongoEnableSSL;
```

Comment:
How about:
boolean enableSSL = Boolean.TRUE.equals(mongoEnableSSL);
?
Or even better use Boolean.TRUE.equals(mongoEnableSSL); on the setAllProperties parameter directly below, instead of creating a new variable.